server/chat/backend/agent/tools/gitlab_tool.py

```
@@ -0,0 +1,576 @@
+            "files_committed": [f["file_path"] for f in terraform_files],
+        })
+    except Exception as e:
+        logger.error(f"Error in commit_terraform: {e}", exc_info=True)
+        return json.dumps({"status": "error", "error": f"GitLab commit failed: {str(e)}"})
+
+
+# ---------------------------------------------------------------------------
+# Main entry point
+# ---------------------------------------------------------------------------
+
+def gitlab_tool(
```

Ruff (0.15.12)

[warning] Too many return statements (10 > 6)

(PLR0911)